python/src/_dynamic_memory_index.py

```diff
@@ -302,7 +307,21 @@ def save(self, save_path: str, compact_before_save: bool = True):
         """
         if save_path == "":
             raise ValueError("save_path cannot be empty")
-        self._index.save(save_path=save_path, compact_before_save=compact_before_save)
+        if index_prefix == "":
+            raise ValueError("index_prefix cannot be empty")
+        _assert_existing_directory(save_path, "save_path")
+        _save_path = os.path.join(save_path, index_prefix)
```

Comment:
Honestly, there's no reason for this to not be save_path anyway - I use the `_` prefix without `self.` for when it's something like `Optional[str]` and I've already done the "this is def. not None so I can access it" without mypy being angry that I did it in a way it didn't expect me to.  It can get cranky if you change the type of something from "Optional[str]" to definitely "str".

So really it can just be save_path.